lib/quicly.c

```diff
@@ -408,27 +417,31 @@ static const quicly_transport_parameters_t default_transport_params = {.max_udp_
                                                                        .max_ack_delay = QUICLY_DEFAULT_MAX_ACK_DELAY,
                                                                        .min_ack_delay_usec = UINT64_MAX};
 
-static __thread int64_t now;
-static __thread size_t in_quicly_send_cnt;
-
-static void update_now(quicly_context_t *ctx)
+static void lock_now(quicly_conn_t *conn, int is_reentrant)
 {
-    if (in_quicly_send_cnt == 0) {
-        int64_t newval = ctx->now->cb(ctx->now);
-        if (now < newval)
-            now = newval;
+    if (conn->stash.now == 0) {
```

Comment:
maybe add `assert(conn->stash.lock_count == 0)` ?

Reply:
Good point! Done in 80baf1d.